cmd/kubeadm/app/componentconfigs/defaults.go

```diff
@@ -72,6 +74,14 @@ func DefaultKubeletConfiguration(internalcfg *kubeadmapi.ClusterConfiguration) {
 		Scheme.Convert(internalcfg.ComponentConfigs.Kubelet, externalkubeletcfg, nil)
 	}
 
+	execer := utilsexec.New()
+	driver, err := kubeadmutil.GetCgroupDriverDocker(execer)
```

Comment:
This code needs to be executed only if the container runtime is Docker. It would break containerd, CRI-O, etc.
To do that we need to get the `NodeRegistrationOptions` propagated to this func as an additional parameter as well.

Reply:
^ yes, this is true.